docs/launch-behavior.md

```diff
@@ -0,0 +1,165 @@
+| `atom /b/3.md`              | [/a 3.md]      | [/a] [ 3.md] :warning: | [/a 3.md] :bug:    |
+| `atom /b`                   | [/a] [/b]      | [/a] [/b]              | [/a] [/b]          |
+| `atom --add /a/1.md`        | [/a 1.md]      | [/a 1.md]              | [/a 1.md]          |
+| `atom --add /a`             | [/a]           | [/a]                   | [/a]               |
+| `atom --add /b/3.md`        | [/a,/b 3.md]   | [/a 3.md] :warning:    | [/a 3.md] :new:    |
+| `atom --add /b`             | [/a,/b]        | [/a,/b]                | [/a,/b]            |
+| `atom --new-window /a/1.md` | [/a] [/a 1.md] | [/a] [ 1.md] :warning: | [/a] [ 1.md] :new: |
+| `atom --new-window /a`      | [/a] [/a]      | [/a] [/a]              | [/a] [/a]          |
+| `atom --new-window /b/3.md` | [/a] [/b 3.md] | [/a] [ 3.md] :warning: | [/a] [ 3.md] :new: |
+| `atom --new-window /b`      | [/a] [/b]      | [/a] [/b]              | [/a] [/b]          |
+
+### Open windows, one with a project root and one without
+
+> [/a] []
+
+| Action | Outcome (<=1.35.1) | Outcome (1.36.0) | Outcome (1.36.1) |
+| --- | --- | --- | --- |
+| `atom /a/1.md`              | [/a 1.md] []      | [/a 1.md] []              | [/a 1.md] []             |
+| `atom /a`                   | [/a] []           | [/a] []                   | [/a] []                  |
+| `atom /b/3.md`              | [/a] [/b 3.md]    | [/a] [] [ 3.md] :warning: | [/a] [ 3.md] :bug: :new: |
+| `atom /b`                   | [/a] [/b]         | [/a] [] [/b] :warning:    | [/a] [/b] :bug:          |
+| `atom --add /a/1.md`        | [/a 1.md] []      | [/a 1.md] []              | [/a 1.md] []             |
+| `atom --add /a`             | [/a] []           | [/a] []                   | [/a] []                  |
+| `atom --add /b/3.md`        | [/a] [/b 3.md]    | [/a] [ 3.md] :warning:    | [/a] [ 3.md] :new:       |
+| `atom --add /b`             | [/a] [/b]         | [/a] [/b]                 | [/a] [/b]                |
+| `atom --new-window /a/1.md` | [/a] [] [/a 1.md] | [/a] [] [ 1.md] :warning: | [/a] [] [ 1.md] :new:    |
+| `atom --new-window /a`      | [/a] [] [/a]      | [/a] [] [/a]              | [/a] [] [/a]             |
+| `atom --new-window /b/3.md` | [/a] [] [/b 3.md] | [/a] [] [ 3.md] :warning: | [/a] [] [ 3.md] :new:    |
+| `atom --new-window /b`      | [/a] [] [/b]      | [/a] [] [/b]              | [/a] [] [/b]             |
+
+> [] [/a]
+
+| Action | Outcome (<=1.35.1) | Outcome (1.36.0) | Outcome (1.36.1) |
+| --- | --- | --- | --- |
+| `atom /a/1.md`              | [] [/a 1.md]      | [] [/a 1.md]              | [] [/a 1.md]          |
+| `atom /a`                   | [] [/a]           | [] [/a]                   | [] [/a]               |
+| `atom /b/3.md`              | [] [/a 3.md]      | [] [/a] [ 3.md] :warning: | [] [/a 3.md] :bug:    |
+| `atom /b`                   | [] [/a] [/b]      | [] [/a] [/b]              | [/b] [/a] :bug: :new: |
```

Comment:
This change feels like a "nice-to-have" to me. If we can make it work easily, then I think it's a nice improvement. Otherwise, I recommend that we defer it for a future pull request, since this change wasn't an explicit goal of #18608.

Reply:
👍 Makes sense to me. I _think_ it'll actually be easier to implement this than preserving the old inconsistent behavior, but I won't know for sure until I'm all up in AtomApplication again.